scripts/skills/perks/perk_rf_dismemberment.nut

```diff
@@ -0,0 +1,71 @@
+this.perk_rf_dismemberment <- ::inherit("scripts/skills/skill", {
+	m = {
+		IsForceEnabled = false,
+		NumInjuriesBefore = 0
+	},
+	function create()
+	{
+		this.m.ID = "perk.rf_dismemberment";
+		this.m.Name = ::Const.Strings.PerkName.RF_Dismemberment;
+		this.m.Description = ::Const.Strings.PerkDescription.RF_Dismemberment;
+		this.m.Icon = "ui/perks/rf_dismemberment.png";
+		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Order = ::Const.SkillOrder.Perk;
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsHidden = false;
+	}
+
+	// Get all the possible injuries, filter out any that are already present on the target or are excluded from the target
+	// Then adjust the InjuryThresholdMult so that the highest injury's threshold ends up being that of the lowest possible injury in the original injury pool
+	// This ensures that the injury inflicted is always one of the highest possible ones
+	function onBeforeTargetHit( _skill, _targetEntity, _hitInfo )
```

Comment:
I don't understand what's going on in this function even after reading the comments and the perk description.
Got any examples on how this is supposed to function? And what it's wanting to achieve?

Reply:
- Injuries A, B, C have a threshold of 25%. This means if you do 25% damage you will inflict one of these injuries.
- Injuries D, E, F have a threshold of 50%. This means if you do 50% damage you will inflict A, B, C, D, E, or F injury.
- What this perk does is that it changes the `InjuryThresholdMult` by 25 / 50 = 0.5 times and then removes all the 25% injuries from the pool. This makes the effective injury threshold of D, E, F as 50 * 0.5 = 25%. So as long as you did enough damage to inflict the lowest possible injury from the pool, you will inflict one of the highest possible injuries.

Reply:
In that case I find a clarification in the perk description in order.
I though that it was just smart injuries which cull the lowest threshold, once you entered the next one.
But what you do here is upgrade any inflicted injury into the highest possible one.

`Injuries inflicted with attacks are upgraded into one with the highest injury threshold.`
or
`Whenever you would inflict an injury with an attack, instead inflict an applicable injury with the highest injury`threshold.´

Reply:
That's a good point. How about `When inflicting injuries, always inflict one with the highest damage threshold, even if your damage did not meet that threshold.`?

Reply:
That's fine by me